presto-docs/src/main/sphinx/security/authorization.rst

```diff
@@ -0,0 +1,130 @@
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following is an example of the properties that need to be added to the
+``config.properties`` file:
+
+.. code-block:: none
+
+    http-server.authorization.enabled=true
+    http-server.authorization.default-policy=ALLOW
+    http-server.authorization.default-allowed-roles=USER,ADMIN
+    http-server.authorization.allow-unsecured-requests=false
+
+======================================================= ======================================================
+Property                                                Description
+======================================================= ======================================================
+``http-server.authorization.enabled``                   Enable authorization for the Presto.
+                                                        Should be set to ``true``. Default value is
+                                                        ``false``.
+``http-server.authorization.default-policy``            The default authorization policy applies to endpoints
+                                                        without allowed roles specified. Can be set to
+                                                        ``ALLOW``, ``DENY`` and ``DEFAULT_ROLES``.
+``http-server.authorization.default-allowed-roles``     The roles allowed to access the endpoints without
+                                                        explicitly specified when default-policy is set to
+                                                        ``DEFAULT_ROLES``.
+``http-server.authorization.allow-unsecured-requests``  Skip authorization check for unsecured requests.
+                                                        Default value is ``false``.
+======================================================= ======================================================
+
+.. warning::
+
+    Enable allowing unsecured requests only when you leave HTTP enabled and
+    need the HTTP endpoints to be accessible. In most cases this is a security
+    hole.
+
+
+Authorizer Configuration
+------------------------
+
+To enable authorization, the interface
+``com.facebook.airlift.http.server.Authorizer`` must be implemented and bound.
+It performs the actual authorization check based on the principal of incoming
+request and the allowed roles of endpoint being requested, determines if the
+principal belongs to at least one of the allowed roles.
+
+Configuration-based Authorizer
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This plugin allows you to turn on authorization support by specifying an
+identity regex file. To use this plugin, add an ``etc/roles.properties`` file
+containing regexes for each allowed role.
+
+Identity Regex
+~~~~~~~~~~~~~~~~~~~~~
+
+The principal is granted role(s) based on the matching regex read from the
+identity regex file. If no role matches, access is denied. There are three
+roles defined in Presto:
+
+* ``user``: who would send HTTP request from external such as client.
```

Comment:
Users who should have access to external endpoints like those needed to launch queries, check status and get output data and provides data for UI.